azure-cosmos/src/main/java/io/micronaut/azure/cosmos/client/CosmosClientFactory.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2017-2020 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.azure.cosmos.client;
+
+import com.azure.cosmos.CosmosAsyncClient;
+import com.azure.cosmos.CosmosClient;
+import io.micronaut.context.annotation.Bean;
+import io.micronaut.context.annotation.Factory;
+import io.micronaut.context.annotation.Requires;
+import io.micronaut.core.annotation.Internal;
+
+/**
+ * The Azure Cosmos Client factory.
+ *
+ * @author radovanradic
+ * @since 3.5.0
+ */
+@Factory
+@Internal
+final class CosmosClientFactory {
```

Comment:
Any reason why this is final and not public. Also, why `@Internal`?

In general, all those things will prevent users from replacing/subclassing our factory

Reply:
It will also prevent us from modifying it. IMHO there should be another way to do it without relying on our implementation

Reply:
A new factory with an order should be the way to go